src/js/addons/widgets/Widget.js

```diff
@@ -1,34 +1,62 @@
+/**
+ *  A widget is a standalone unit of functionality for a
+ *  given matching context in Tangram YAML.
+ *
+ *  An instance of the widget class stores a reference to
+ *  its DOM element (and children) and can be extended
+ *  with any additional functionality.
+ *
+ */
+
+import { getPosition } from '../../core/common.js';
+
+// Store reference to the editor on this module
+let editor;
```

Comment:
So, you're right, let's keep things simple. I will refer to the global tangramPlay for now and avoid having to pass a reference to the editor down through every function call.